src/legacy/core_plugins/vis_default_editor/public/components/controls/include_exclude.tsx

```diff
@@ -0,0 +1,146 @@
+  const [numbers, setNumbers] = useState(
+    getInitModelList(
+      props.value && isArray(props.value) ? (props.value as Array<number | undefined>) : [undefined]
+    )
+  );
+  const numberRange = useMemo(() => getRange('[-Infinity,Infinity]'), []);
+  const [isValid, setIsValid] = useState(true);
+
+  const setValidity = useCallback(
+    (isListValid: boolean) => {
+      setIsValid(isListValid);
+      props.setValidity(isListValid);
+    },
+    [props.setValidity]
+  );
+
+  useValidation(setValidity, !hasInvalidValues(numbers));
+
+  const onUpdate = useCallback((numberList: NumberRowModel[]) => {
+    setNumbers(numberList);
+    props.setValue(
+      numberList.map(({ value }) => value).filter(value => value !== EMPTY_STRING) as number[]
+    );
+  }, []);
+
+  const onChangeValue = useCallback(
+    ({ id, value }: { id: string; value: string }) => {
+      const parsedValue = parse(value);
+
+      onUpdate(
+        numbers.map(number =>
+          number.id === id
+            ? {
+                id,
+                value: parsedValue,
+                isInvalid: value !== EMPTY_STRING && isNaN(parseFloat(value)),
+              }
+            : number
+        )
+      );
+    },
+    [numberRange, numbers, onUpdate]
+  );
+
+  // Add an item to the end of the list
+  const onAdd = useCallback(() => {
+    const newArray = [
+      ...numbers,
+      {
+        id: generateId(),
+        value: '',
+        isInvalid: false,
+      } as NumberRowModel,
+    ];
+    onUpdate(newArray);
+  }, [numbers, numberRange, onUpdate]);
+
+  const onDelete = useCallback(
+    (id: string) => {
+      const newArray = numbers.filter(model => model.id !== id);
+      onUpdate(newArray);
+    },
+    [numbers, onUpdate]
+  );
+
+  return isNumberType(props.agg) ? (
+    <EuiFormRow
+      id={`${props.aggParam.name}-${props.agg.id}}`}
+      label={props.aggParam.displayName || props.aggParam.name}
+      fullWidth={true}
+      compressed
+      isInvalid={props.showValidation ? !isValid : false}
+    >
+      <>
+        {numbers.map((number, arrayIndex) => (
+          <Fragment key={number.id}>
+            <NumberRow
+              isInvalid={number.isInvalid}
+              disableDelete={numbers.length === 1}
+              model={number}
+              labelledbyId={`${props.aggParam.name}-${props.agg.id}-legend`}
+              range={numberRange}
+              onDelete={onDelete}
+              onChange={onChangeValue}
+              onBlur={props.setTouched}
+              autoFocus={numbers.length !== 1 && arrayIndex === numbers.length - 1}
+            />
+            {numbers.length - 1 !== arrayIndex && <EuiSpacer size="s" />}
+          </Fragment>
+        ))}
+        <EuiSpacer size="s" />
+        <EuiFlexItem>
+          <EuiButtonEmpty iconType="plusInCircleFilled" onClick={onAdd} size="xs">
+            <FormattedMessage
+              id={`visDefaultEditor.controls.${props.aggParam.name}.addUnitButtonLabel`}
```

Comment:
For i18n IDs static strings must be used:
```suggestion
              id="visDefaultEditor.controls.includeExclude.addUnitButtonLabel"
```